package-deps/src/tests/getPackageDeps.test.ts

```diff
@@ -0,0 +1,177 @@
+      try {
+        const expectedFiles: { [key: string]: string } = {
+          'a.txt': '2e65efe2a145dda7ee51d1741299f848e5bf752e',
+          'file1.txt': 'c7b2f707ac99ca522f965210a7b6b0b109863f34',
+          'package.json': '33703d582243a41bdebff8ee7dd046a01fc054b9'
+        };
+        const filePaths: string[] = Object.keys(results.files).sort();
+
+        filePaths.forEach(filePath => (
+          expect(
+            results.files[filePath])
+              .equals(expectedFiles[filePath], `path: ${filePath}`)));
+
+      } catch (e) {
+        return _done(e);
+      }
+
+      _done();
+    });
+
+  });
+
+  it('can can handle removing one file', (done) => {
+    const testFilePath: string = path.join(TEST_PROJECT_PATH, 'file1.txt');
+
+    fs.unlinkSync(testFilePath);
+
+    function _done(e?: Error): void {
+      execSync(`git checkout ${ testFilePath }`);
+      done(e);
+    }
+
+    getPackageDeps(TEST_PROJECT_PATH).then((results) => {
+      try {
+        const expectedFiles: { [key: string]: string } = {
+          'package.json': '33703d582243a41bdebff8ee7dd046a01fc054b9'
+        };
+        const filePaths: string[] = Object.keys(results.files).sort();
+
+        filePaths.forEach(filePath => (
+          expect(results.files[filePath])
+            .equals(expectedFiles[filePath], `path: ${filePath}`)));
+
+      } catch (e) {
+        return _done(e);
+      }
+
+      _done();
+    });
+  });
+
+  it('can can handle changing one file', (done) => {
+    const testFilePath: string = path.join(TEST_PROJECT_PATH, 'file1.txt');
+
+    fs.writeFileSync(testFilePath, 'abc');
+
+    function _done(e?: Error): void {
+      execSync(`git checkout ${testFilePath}`);
+      done(e);
+    }
+
+    getPackageDeps(TEST_PROJECT_PATH).then((results) => {
+      try {
+        const expectedFiles: { [key: string]: string } = {
+          'file1.txt': 'f2ba8f84ab5c1bce84a7b441cb1959cfc7093b7f',
+          'package.json': '33703d582243a41bdebff8ee7dd046a01fc054b9'
+        };
+        const filePaths: string[] = Object.keys(results.files).sort();
+
+        filePaths.forEach(filePath => (
+          expect(results.files[filePath])
+            .equals(expectedFiles[filePath], `path: ${filePath}`)));
+
+      } catch (e) {
+        return _done(e);
+      }
+
+      _done();
+    });
+  });
+
+  it('can exclude a committed file', (done) => {
+    getPackageDeps(TEST_PROJECT_PATH, [ 'file1.txt' ]).then((results) => {
+      try {
+        const expectedFiles: { [key: string]: string } = {
+          'package.json': '33703d582243a41bdebff8ee7dd046a01fc054b9'
+        };
+        const filePaths: string[] = Object.keys(results.files).sort();
+
+        filePaths.forEach(filePath => (
+          expect(results.files[filePath])
+            .equals(expectedFiles[filePath], `path: ${filePath}`)));
+
+      } catch (e) { return done(e); }
+
+      done();
+    });
+  });
+
+  it('can exclude an added file', (done) => {
```

Comment:
I guess you can possibly write edge case handling tests too (such as what happens when you exclude a file that does not exist, aka exclude has a random string).